packages/supabase_flutter/lib/src/local_storage.dart

```diff
@@ -93,31 +87,129 @@ class HiveLocalStorage extends LocalStorage {
     await Hive.openBox(_hiveBoxName, encryptionCipher: encryptionCipher);
   }
 
-  static Future<bool> _hasAccessToken() {
+  @override
+  Future<bool> hasAccessToken() {
     return Future.value(
       Hive.box(_hiveBoxName).containsKey(
         supabasePersistSessionKey,
       ),
     );
   }
 
-  static Future<String?> _accessToken() {
+  @override
+  Future<String?> accessToken() {
     return Future.value(
       Hive.box(_hiveBoxName).get(supabasePersistSessionKey) as String?,
     );
   }
 
-  static Future<void> _removePersistedSession() {
+  @override
+  Future<void> removePersistedSession() {
     return Hive.box(_hiveBoxName).delete(supabasePersistSessionKey);
   }
 
-  static Future<void> _persistSession(String persistSessionString) {
+  @override
+  Future<void> persistSession(String persistSessionString) {
     // Flush after X amount of writes
     return Hive.box(_hiveBoxName)
         .put(supabasePersistSessionKey, persistSessionString);
   }
 }
 
+/// A [LocalStorage] implementation that implements SharedPreferences as the
+/// storage method.
+class SharedPreferencesLocalStorage extends LocalStorage {
+  late final SharedPreferences _prefs;
+
+  @override
+  Future<void> initialize() async {
+    WidgetsFlutterBinding.ensureInitialized();
+    _prefs = await SharedPreferences.getInstance();
+  }
+
+  @override
+  Future<bool> hasAccessToken() {
+    return Future.value(_prefs.containsKey(supabasePersistSessionKey));
```

Comment:
Could we maybe use the [same storage key as the js client library](https://github.com/supabase/gotrue-js/blob/master/src/lib/constants.ts#L3)?
Related https://github.com/supabase/supabase-flutter/issues/453

Reply:
I still don't think this enables both native web and flutter to work together. Shared Prefs stores them in localstorage and native web uses cookies (Though I'm not exactly sure how they work).

Reply:
@Vinzent03 The JS sdk stores them in a local storage, and there are [auth-helpers](https://supabase.com/docs/guides/auth/auth-helpers) that takes the session and puts them in cookies. No need to worry about the cookie part, but no sense in having a different storage keys compared to the JS library. Let's match it.